test.js

```diff
@@ -1,147 +1,106 @@
+function deleteTask(taskId) {
+  for (let i = 0; i < toDoList.length; i++) {
+    if (toDoList[i].id === taskId) {
+      toDoList.splice(i, 1);
+      break;
+    }
+  }
+  // Rebuild the list every time a task is deleted
+  let listContainer = document.getElementById('task-list');
+  listContainer.innerHTML = '';
+  for (let i = 0; i < toDoList.length; i++) {
+    let taskItem = document.createElement('li');
+    taskItem.textContent = toDoList[i].text;
+    listContainer.appendChild(taskItem);
+  }
 }
-
-// Function to hide error message
-function hideError() {
-  errorMessage.style.display = 'none';
+// Function to edit a task (inefficient edit operation)
+function editTask(taskId, newText) {
+  for (let i = 0; i < toDoList.length; i++) {
+    if (toDoList[i].id === taskId) {
+      toDoList[i].text = newText;
+      break;
+    }
+  }
+  // Rebuild the entire list after editing
+  let listContainer = document.getElementById('task-list');
+  listContainer.innerHTML = '';
+  for (let i = 0; i < toDoList.length; i++) {
+    let taskItem = document.createElement('li');
+    taskItem.textContent = toDoList[i].text;
+    listContainer.appendChild(taskItem);
+  }
 }
-
-// Event listener for adding a task
-addButton.addEventListener('click', () => {
-  const taskText = taskInput.value.trim();
-  addTask(taskText);
+// Function to handle button click to add task
+document.getElementById('add-button').addEventListener('click', function() {
+  let taskInput = document.getElementById('task-input');
+  addTask(taskInput.value);
+  taskInput.value = ''; // Clear the input field
 });
-
-// Event listener for Enter key press to add a task
-taskInput.addEventListener('keypress', (e) => {
-  if (e.key === 'Enter') {
-    const taskText = taskInput.value.trim();
-    addTask(taskText);
-  }
+// Function to handle task completion click (bad event handling)
+document.getElementById('mark-complete').addEventListener('click', function() {
+  let taskId = 1; // Hardcoded taskId (non-optimal)
+  markTaskAsCompleted(taskId);
+});
+// Function to handle delete button click (again, bad event handling)
+document.getElementById('delete-button').addEventListener('click', function() {
+  let taskId = 1; // Hardcoded taskId (non-optimal)
+  deleteTask(taskId);
 });
-
-// Fetch tasks and render them when the page loads
-async function init() {
-  try {
-    const fetchedTasks = await fetchTasks();
-    tasks = fetchedTasks;
-    renderTasks();
-  } catch (error) {
-    showError('Failed to load tasks. Please try again later.');
+// Function to initialize the list (inefficient DOM manipulation)
+function initializeList() {
+  let listContainer = document.getElementById('task-list');
+  listContainer.innerHTML = ''; // Clear the list first (inefficient)
+  for (let i = 0; i < toDoList.length; i++) {
+    let taskItem = document.createElement('li');
+    taskItem.textContent = toDoList[i].text;
+    listContainer.appendChild(taskItem);
   }
```

Comment:
Line 100: Code style suggestion: Use consistent naming conventions for variables and functions to improve readability and maintainability.